frontend/src/utils/DeviceManager/deviceManager.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Device manager that retrieves the stored device ID for a given device type (audio or video)
+ * and checks if it is still connected.
+ */
+export default class DeviceManager {
+  private devices: MediaDeviceInfo[] = [];
+
+  async init() {
+    if (navigator.mediaDevices) {
+      this.devices = await navigator.mediaDevices.enumerateDevices();
+    }
+  }
+
+  getConnectedDeviceId(kind: 'audioinput' | 'videoinput'): string | undefined {
+    const key = kind === 'videoinput' ? 'videoSource' : 'audioSource';
+    const storedId = window.localStorage.getItem(key);
+    return this.devices.find((device) => device.kind === kind && device.deviceId === storedId)
```

Comment:
Do we need this filter?  Naively, seems like we can just store the `deviceId` in `localStorage` as well.

Reply:
Yes, this is meant to check if the deviceId that was previously stored in the `localStorage` is still connected to the device and has not been unplugged 🙏